CodeEdit/Features/CodeEditUI/Views/IconButtonStyle.swift

```diff
@@ -0,0 +1,126 @@
+//
+//  IconButtonStyle.swift
+//  CodeEdit
+//
+//  Created by Austin Condiff on 11/9/23.
+//
+
+import SwiftUI
+
+struct IconButtonStyle: ButtonStyle {
+    var isActive: Bool?
+    var font: Font?
+    var size: CGSize?
+
+    init() {
+        self.isActive = nil
+        self.font = nil
+        self.size = nil
+    }
+
+    init(isActive: Bool? = nil, font: Font? = nil, size: CGFloat? = nil) {
+        self.isActive = isActive
+        self.font = font
+        self.size = size == nil ? nil : CGSize(width: size ?? 0, height: size ?? 0)
+    }
+
+    init(isActive: Bool? = nil, font: Font? = nil, size: CGSize? = nil) {
+        self.isActive = isActive
+        self.font = font
+        self.size = size
+    }
+
+    init(isActive: Bool? = nil, font: Font? = nil) {
+        self.isActive = isActive
+        self.font = font
+        self.size = nil
+    }
```

Comment:
Using default parameters you can combine these init functions into two

```swift
init(isActive: Bool? = nil, font: Font? = nil, size: CGFloat? = nil) {
    self.isActive = isActive
    self.font = font
    if let size = size {
        self.size = CGSize(width: size, height: size)
    } else {
        self.size = nil
    }
}

init(isActive: Bool? = nil, font: Font? = nil, size: CGSize? = nil) {
    self.isActive = isActive
    self.font = font
    self.size = size
}
```

Reply:
Again, I do that and I get this error...
<img width="707" alt="image" src="https://github.com/CodeEditApp/CodeEdit/assets/806104/91d16dd6-75d7-4960-85bc-98f84b65e5a2">

Reply:
So instead let's remove this function:
```
init() {
        self.isActive = nil
        self.font = nil
        self.size = nil
    }
```

I confirmed it works without it.